File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
      int n = s.size();
      int m = t.size();
      if(m!=n){
        return false;
      }
      vector<int> arr(26,0);
      for(int i=0;i<n;i++){
        arr[s[i] - 'a']++;
      }
      for(int j=0;j<m;j++){
        arr[t[j] - 'a']--;
      }
      for(int k=0;k<arr.size();k++){
        if(arr[k]!=0){
          return false;
        }
      }
      return true;
    }
};

File: Data Structures & Algorithms/find-k-closest-elements/submission-0.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        n=len(arr)
        if k==n:
            return arr
        left = 0
        right = n-1
        while left<right:
            mid = (left+right)// 2
            
            if arr[mid]<x:
                left = mid+1
            else:
                right = mid
        a = left-1
        b = left
        
        i = 0
        ans = []
        while i<k :

            if b>=n or (a>=0 and (abs(arr[a] - x) <= abs(arr[b] - x))):
                ans.append(arr[a])
                a-=1
            else:
                ans.append(arr[b])
                b+=1
            i+=1
        ans.sort()
        return ans
            


                    